apps/system/init/init.c

```diff
@@ -157,5 +161,17 @@ int preapp_start(int argc, char *argv[])
 	}
 #endif
 
+/***********************************************************************************
+ *	current preapp_start does the up_cxxinitialize which initializes the
+ *	static constructors. All the tasks and threads created by preapp refer to
+ *	these constructors. To make sure that c++ test cases from tash runs properly,
+ *	we need to keep preapp/constructor initializations alive with below changes.
+ **********************************************************************************/
+#ifdef CONFIG_HAVE_CXXINITIALIZE
+	while(1)
+	{
+		sleep(10);
```

Comment:
Is there any other way to stay in inactive state without wake up at every 10 secs?

Reply:
I will change to use sem wait